public/app/plugins/panel/table/module.ts

```diff
@@ -118,6 +121,24 @@ class TablePanelCtrl extends MetricsPanelCtrl {
   render() {
     this.table = transformDataToTable(this.dataRaw, this.panel);
     this.table.sort(this.panel.sort);
+
+    for (let colIndex = 0; colIndex < this.table.columns.length; colIndex++) {
+      let column = this.table.columns[colIndex];
+
+      for (let i = 0; i < this.panel.styles.length; i++) {
+        let style = this.panel.styles[i];
+        var regex = kbn.stringToJsRegex(style.pattern);
+        const matches = column.text.match(regex);
+        if (matches) {
+          column.style = style;
+          if (style.alias) {
+            column.title = column.text.replace(regex, style.alias);
+          }
+          break;
+        }
+      }
+    }
```

Comment:
this block of code would be great to have some tests for, and some docs as it includes some unobvious regex replace features.